BackendWebApp/src/main/java/uk/ac/ebi/biostd/webapp/application/security/service/MagicFolderUtil.java

```diff
@@ -0,0 +1,52 @@
+
+package uk.ac.ebi.biostd.webapp.application.security.service;
+
+import static java.lang.String.format;
+import static java.nio.file.attribute.PosixFilePermissions.fromString;
+
+import java.io.File;
+import java.nio.file.Files;
+import lombok.SneakyThrows;
+import org.springframework.stereotype.Component;
+import uk.ac.ebi.biostd.webapp.application.configuration.ConfigProperties;
+
+@Component
+public class MagicFolderUtil {
```

Comment:
Since the logic behind the folder creation is so obscure right now (hopefully we'll improve it in the future), I think it would be nice (if possible) to have at least a couple of examples of how this secret/magicKey combination works.